contrib/gorilla/websocket/websocket_test.go

```diff
@@ -0,0 +1,158 @@
+				mt, msg, err := conn.ReadMessage()
+				require.NoError(t, err)
+				require.Equal(t, websocket.TextMessage, mt)
+				require.Equal(t, []byte("message"), msg)
+			},
+			server: func(conn *websockettrace.Conn) {
+				err := conn.WriteMessage(websocket.TextMessage, []byte("message"))
+				require.NoError(t, err)
+			},
+			assert: func(mt mocktracer.Tracer) {
+				require.Empty(t, mt.OpenSpans())
+
+				spans := mt.FinishedSpans()
+				require.Len(t, spans, 2)
+
+				span := spans[0]
+				require.Equal(t, "websocket.write_message", span.OperationName())
+				require.Equal(t, len("message"), span.Tag("websocket.message_length"))
+				require.Equal(t, websocket.TextMessage, span.Tag("websocket.message_type"))
+
+				span = spans[1]
+				require.Equal(t, "websocket.read_message", span.OperationName())
+			},
+		},
+		{
+			name: "WritePrepareMessage",
+			client: func(conn *websockettrace.Conn) {
+				mt, msg, err := conn.ReadMessage()
+				require.NoError(t, err)
+				require.Equal(t, websocket.TextMessage, mt)
+				require.Equal(t, []byte("message"), msg)
+			},
+			server: func(conn *websockettrace.Conn) {
+				msg, err := websocket.NewPreparedMessage(websocket.TextMessage, []byte("message"))
+				err = conn.WritePreparedMessage(msg)
+				require.NoError(t, err)
+			},
+			assert: func(mt mocktracer.Tracer) {
+				require.Empty(t, mt.OpenSpans())
+
+				spans := mt.FinishedSpans()
+				require.Len(t, spans, 2)
+
+				span := spans[0]
+				require.Equal(t, "websocket.write_prepared_message", span.OperationName())
+
+				span = spans[1]
+				require.Equal(t, "websocket.read_message", span.OperationName())
+			},
+		},
+		{
+			name: "WriteJSON/ReadJSON",
+			client: func(conn *websockettrace.Conn) {
+				var s string
+				err := conn.ReadJSON(&s)
+				require.NoError(t, err)
+				require.Equal(t, "message", s)
+			},
+			server: func(conn *websockettrace.Conn) {
+				err := conn.WriteJSON("message")
+				require.NoError(t, err)
+			},
+			assert: func(mt mocktracer.Tracer) {
+				require.Empty(t, mt.OpenSpans())
+
+				spans := mt.FinishedSpans()
+				require.Len(t, spans, 2)
+
+				span := spans[0]
+				require.Equal(t, "websocket.write_json", span.OperationName())
+
+				span = spans[1]
+				require.Equal(t, "websocket.read_json", span.OperationName())
+			},
+		},
+		{
+			name:   "WriteControl",
+			client: func(*websockettrace.Conn) {},
+			server: func(conn *websockettrace.Conn) {
+				err := conn.WriteControl(websocket.PingMessage, []byte("message"), time.Now().Add(time.Second))
+				require.NoError(t, err)
+			},
+			assert: func(mt mocktracer.Tracer) {
+				require.Empty(t, mt.OpenSpans())
+
+				spans := mt.FinishedSpans()
+				require.Len(t, spans, 1)
+
+				span := spans[0]
+				require.Equal(t, "websocket.write_control", span.OperationName())
+				require.Equal(t, len("message"), span.Tag("websocket.message_length"))
+				require.Equal(t, websocket.PingMessage, span.Tag("websocket.message_type"))
+			},
+		},
+	} {
+		tc := tc
```

Comment:
Let's remove this 🙂

Reply:
Indeed we can avoid it because the test isn't run in parallel, so the following closure will be okay when synchronously executed.

But since this is a very common mistake that can be pretty hard to find, I tend to always define a scope there before having closures on them and avoid any issue.

Let me know what you really prefer =)